openedx/core/djangoapps/theming/helpers.py

```diff
@@ -52,3 +71,311 @@ def get_themed_template_path(relative_path, default_path, **kwargs):
+        root_name,
+        "templates"
+    ])
+
+    # strip `/` if present at the start of relative_path
+    template_name = re.sub(r'^/+', '', relative_path)
+    search_path = os.path.join(template_path, template_name)
+    if os.path.isfile(search_path):
+        path = '/{site_theme_dir}/{root_name}/templates/{template_name}'.format(
+            site_theme_dir=site_theme_dir,
+            root_name=root_name,
+            template_name=template_name,
+        )
+        return path
+    else:
+        return relative_path
+
+
+def get_current_theme_template_dirs():
+    """
+    Returns template directories for the current theme.
+
+    Example:
+        >> get_current_theme_template_dirs('header.html')
+        ['/edx/app/edxapp/edx-platform/themes/red-theme/lms/templates/', ]
+
+    Returns:
+        (list): list of directories containing theme templates.
+    """
+    site_theme_dir = get_current_site_theme_dir()
+    if not site_theme_dir:
+        return None
+
+    base_theme_dir = get_base_theme_dir()
+    root_name = get_project_root_name()
+    template_path = "/".join([
+        base_theme_dir,
+        site_theme_dir,
+        root_name,
+        "templates"
+    ])
+
+    return [template_path]
+
+
+def strip_site_theme_templates_path(uri):
+    """
+    Remove site template theme path from the uri.
+
+    Example:
+        >> strip_site_theme_templates_path('/red-theme/lms/templates/header.html')
+        'header.html'
+
+    Arguments:
+        uri (str): template path from which to remove site theme path. e.g. '/red-theme/lms/templates/header.html'
+
+    Returns:
+        (str): template path with site theme path removed.
+    """
+    site_theme_dir = get_current_site_theme_dir()
+    if not site_theme_dir:
+        return uri
+
+    root_name = get_project_root_name()
+    templates_path = "/".join([
+        site_theme_dir,
+        root_name,
+        "templates"
+    ])
+
+    uri = re.sub(r'^/*' + templates_path + '/*', '', uri)
+    return uri
+
+
+def get_current_site():
+    """
+    Return current site.
+
+    Returns:
+         (django.contrib.sites.models.Site): theme directory for current site
+    """
+    from edxmako.middleware import REQUEST_CONTEXT
+    request = getattr(REQUEST_CONTEXT, 'request', None)
+    if not request:
+        return None
+    return getattr(request, 'site', None)
+
+
+def get_current_site_theme_dir():
+    """
+    Return theme directory for the current site.
+
+    Example:
+        >> get_current_site_theme_dir()
+        'red-theme'
+
+    Returns:
+         (str): theme directory for current site
+    """
+    site = get_current_site()
+    if not site:
+        return None
+    site_theme_dir = cache.get(get_site_theme_cache_key(site))
+
+    # if site theme dir is not in cache and comprehensive theming is enabled then pull it from db.
+    if not site_theme_dir and is_comprehensive_theming_enabled():
+        site_theme = site.themes.first()  # pylint: disable=no-member
```

Comment:
@saleem-latif This line is causing  [WL-377](https://openedx.atlassian.net/browse/WL-377) issue which blocks a CAT-2 [TNL-4155](https://openedx.atlassian.net/browse/TNL-4155)

Can you please have look at this ?